src/models/app_configuration.js

```diff
@@ -54,13 +54,28 @@ async function addLinkedApplication (appData, alias, ignoreParentConfig) {
   return persistConfig(currentConfig);
 };
 
-async function removeLinkedApplication (alias) {
+async function removeLinkedApplication ({ appId, alias }) {
+  function keepPredicated () {
+    if (appId != null) {
+      return (appEntry) => appEntry.app_id !== appId;
+    }
+    if (alias != null) {
+      return (appEntry) => appEntry.alias !== alias;
+    }
+  }
+
   const currentConfig = await loadApplicationConf();
   const newConfig = {
     ...currentConfig,
-    apps: currentConfig.apps.filter((appEntry) => appEntry.alias !== alias),
+    apps: currentConfig.apps.filter(keepPredicated()),
   };
-  return persistConfig(newConfig);
+
+  if (currentConfig.apps.length !== newConfig.apps.length) {
+    await persistConfig(newConfig);
+    return true;
+  }
+
+  return false;
 };
```

Comment:
suggestion: as you know, I'm not a great fan of inline functions declared like this. Also, I'm not sure if it can happen but when `appId` and `alias` are null, `keepPredicated` can return null and will fail when used with `.filter`. What about this:

```js
async function removeLinkedApplication ({ appId, alias }) {

  const currentConfig = await loadApplicationConf();
  const newApps = currentConfig.apps.filter((appEntry) => {
    return appEntry.app_id !== appId && appEntry.alias !== alias;
  });

  if (currentConfig.apps.length !== newApps.length) {
    await persistConfig({
      ...currentConfig,
      apps: newApps,
    });
    return true;
  }

  return false;
};
```